From 631105d38d9aad10e40a7656206467eb9eaabd2b Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <Guoqing.Jiang@windriver.com>
Date: Wed, 9 Jan 2013 15:44:36 +0800
Subject: [PATCH 01/23] fsl_imx53: Add config options and compile support for
 mx53

Refer to vendor-drop package, L2.6.35_11.09.01_ER_source.tar.gz

Signed-off-by: Guoqing Jiang <Guoqing.Jiang@windriver.com>
---
 arch/arm/mach-imx/Kconfig  |    7 +++++++
 arch/arm/mach-imx/Makefile |   12 ++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/arch/arm/mach-imx/Kconfig b/arch/arm/mach-imx/Kconfig
index 7561eca..14ad054 100644
--- a/arch/arm/mach-imx/Kconfig
+++ b/arch/arm/mach-imx/Kconfig
@@ -29,6 +29,7 @@ config ARCH_MX51
 	bool
 
 config ARCH_MX53
+	select ARCH_HAS_CPUFREQ
 	bool
 
 config SOC_IMX1
@@ -808,11 +809,17 @@ config MACH_MX53_SMD
 config MACH_MX53_LOCO
 	bool "Support MX53 LOCO platforms"
 	select SOC_IMX53
+	select IRQ_DOMAIN
+	select IRAM_ALLOC
+	select IMX_HAVE_PLATFORM_MXC_RTC
+	select IMX_HAVE_PLATFORM_MXC_EHCI
 	select IMX_HAVE_PLATFORM_IMX2_WDT
 	select IMX_HAVE_PLATFORM_IMX_I2C
 	select IMX_HAVE_PLATFORM_IMX_UART
 	select IMX_HAVE_PLATFORM_SDHCI_ESDHC_IMX
 	select IMX_HAVE_PLATFORM_GPIO_KEYS
+	select IMX_HAVE_PLATFORM_IMX_SCC
+	select IMX_HAVE_PLATFORM_IMX_DVFS
 	select LEDS_GPIO_REGISTER
 	help
 	  Include support for MX53 LOCO platform. This includes specific
diff --git a/arch/arm/mach-imx/Makefile b/arch/arm/mach-imx/Makefile
index ab939c5..06e1a90 100644
--- a/arch/arm/mach-imx/Makefile
+++ b/arch/arm/mach-imx/Makefile
@@ -76,6 +76,18 @@ ifeq ($(CONFIG_PM),y)
 obj-$(CONFIG_SOC_IMX6Q) += pm-imx6q.o
 endif
 
+ifeq ($(CONFIG_CPU_FREQ_IMX),y)
+obj-$(CONFIG_SOC_IMX53) += cpu_op-mx53.o bus_freq.o
+endif
+
+ifeq ($(CONFIG_USB_EHCI_ARC),y)
+obj-$(CONFIG_SOC_IMX53) += usb_h1.o
+endif
+
+ifeq ($(CONFIG_MXC_PMIC_MC34708),y)
+obj-$(CONFIG_MACH_MX53_LOCO) += mx53_loco_pmic_mc34708.o
+endif
+
 # i.MX5 based machines
 obj-$(CONFIG_MACH_MX51_BABBAGE) += mach-mx51_babbage.o
 obj-$(CONFIG_MACH_MX51_3DS) += mach-mx51_3ds.o
-- 
1.7.9.7

